src/runtime/str.cpp

```diff
@@ -2805,7 +2805,7 @@ void setupStr() {
     str_cls->tp_flags |= Py_TPFLAGS_HAVE_NEWBUFFER;
 
     str_iterator_cls = BoxedClass::create(type_cls, object_cls, &BoxedStringIterator::gcHandler, 0, 0,
-                                          sizeof(BoxedStringIterator), false, "striterator");
+                                          sizeof(BoxedStringIterator), false, "iterator");
```

Comment:
Hmm.. My approach is checking `tp_name` whether should not be subclassable.
As far as I know `striterator ` is not in the Cpython.
so I add `striteator' as should not be suclassable tp_name. 
(This seems to be the case limited to pyston.)